cpp/src/strings/convert/convert_urls.cu

```diff
@@ -325,85 +378,60 @@ std::unique_ptr<column> url_decode(
-  rmm::device_uvector<size_type> esc_positions(esc_count, stream);
-  auto d_esc_positions = esc_positions.data();
-  thrust::copy_if(rmm::exec_policy(stream),
-                  thrust::make_counting_iterator<size_t>(chars_start),
-                  thrust::make_counting_iterator<size_t>(chars_end),
-                  d_esc_positions,
-                  esc_detector);
-
-  // In-place remove any escape positions that crossed string boundaries.
-  device_span<int32_t const> d_offsets_span(d_offsets, offset_count);
-  auto esc_pos_end = thrust::remove_if(rmm::exec_policy(stream),
-                                       d_esc_positions,
-                                       d_esc_positions + esc_count,
-                                       url_decode_esc_position_filter{d_offsets_span});
-
-  // update count in case any were filtered
-  esc_count = esc_pos_end - d_esc_positions;
-  if (esc_count == 0) {
-    // nothing to replace, so just copy the input column
-    return std::make_unique<cudf::column>(strings.parent(), stream, mr);
-  }
-
-  device_span<size_type const> d_esc_positions_span(d_esc_positions, esc_count);
 
   // build offsets column
   auto offsets_column = make_numeric_column(
     data_type{type_id::INT32}, offset_count, mask_state::UNALLOCATED, stream, mr);
-  auto offsets_view = offsets_column->mutable_view();
-  thrust::transform(rmm::exec_policy(stream),
-                    d_offsets_span.begin(),
-                    d_offsets_span.end(),
-                    offsets_view.begin<int32_t>(),
-                    url_decode_offsets_updater{d_esc_positions_span, chars_start});
+
+  // count number of bytes in each string after decoding and store it in offsets_column
+  auto offsets_view         = offsets_column->view();
+  auto offsets_mutable_view = offsets_column->mutable_view();
+  url_decode_char_counter<threadblock_size, block_size>
+    <<<num_threadblocks, threadblock_size, 0, stream.value()>>>(
+      d_in_chars, d_in_offsets, offsets_mutable_view.begin<int32_t>() + 1, strings_count);
+
+  // use inclusive_scan to transform number of bytes into offsets
+  thrust::inclusive_scan(rmm::exec_policy(stream),
+                         offsets_view.begin<int32_t>() + 1,
+                         offsets_view.begin<int32_t>() + offset_count,
+                         offsets_mutable_view.begin<int32_t>() + 1);
+
+  // set the first element of the offset column to 0
+  CUDA_TRY(
+    cudaMemsetAsync(offsets_mutable_view.begin<int32_t>(), 0, sizeof(int32_t), stream.value()));
+
+  // copy the total number of characters of all strings combined (last element of the offset column)
+  // to the host memory
+  int32_t out_chars_bytes;
+  CUDA_TRY(cudaMemcpyAsync(&out_chars_bytes,
+                           offsets_view.begin<int32_t>() + offset_count - 1,
+                           sizeof(int32_t),
+                           cudaMemcpyDeviceToHost,
+                           stream.value()));
+  stream.synchronize();
```

Comment:
This can be done by `cudf::detail::get_value`.

Reply:
Addressed in https://github.com/rapidsai/cudf/pull/8622/commits/f4785201975ef154dfd8d88fe21a04dd9bdde96d.